### ISSUE TITLE
no_yoda_conditions: false positive when using result of method or calculation

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#1954]
```abap
REPORT zfoobar.
DATA var TYPE i.
DATA var2 TYPE i.
IF var = cl_test=>meth( ).
  WRITE 'Test'.
ENDIF.
IF var = var2 + 1.
  WRITE 'Test'.
ENDIF.
```

/zfoobar.prog.abap[4, 4]: No Yoda conditions(no_yoda_conditions)
/zfoobar.prog.abap[7, 4]: No Yoda conditions(no_yoda_conditions)

**Comment by larshp** [comment]
I see both of these as yoda, (not sure if it follows the official definition)

The simple part is on the left of the comparison, and the complex on the right

following is what I'm used to,

```abap
IF cl_test=>meth( ) = var.
IF var2 + 1 = var.
```

**Comment by mbtools** [comment]
Yoda conditions are only about conditions with a constant on one side (https://en.wikipedia.org/wiki/Yoda_conditions). Checking the order based on complexity is something else.

**Comment by larshp** [comment]
yea, still it compromises "readability by increasing the cognitive load of reading the code"

could add a setting "onlyConstants" to the rule

**Comment by larshp** [comment]
alternatively rename the rule to something else, from my POV its about aligning the code and decrease the complexity

**Comment by mbtools** [comment]
onlyConstants makes sense

**Comment by larshp** [comment]
@mbtools try again, 2.74.19

**Comment by mbtools** [comment]
👍
